subprojects/tooling-api/src/main/java/org/gradle/tooling/events/problems/ProblemDescriptor.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.events.problems;
+
+import org.gradle.api.Incubating;
+import org.gradle.tooling.events.OperationDescriptor;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Describes a problem operation.
+ *
+ * @since 8.4
+ */
+@Incubating
+public interface ProblemDescriptor extends OperationDescriptor {
```

Comment:
💭 Might be reordered a bit, so mandatory fields would be on the top and optional ones at the bottom?